sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -473,8 +472,8 @@ class DataFrame private[sql](
         usingColumns.map(col => withPlan(joined.right).resolve(col))
       case FullOuter =>
         usingColumns.map { col =>
-          val leftCol = withPlan(joined.left).resolve(col)
-          val rightCol = withPlan(joined.right).resolve(col)
+          val leftCol = withPlan(joined.left).resolve(col).toAttribute.withNullability(true)
+          val rightCol = withPlan(joined.right).resolve(col).toAttribute.withNullability(true)
           Alias(Coalesce(Seq(leftCol, rightCol)), col)()
```

Comment:
the `leftCol` and `rightCol` are all nullable, but can't be both null right?
so I think the result should be non-nullable